package/src/main/java/com/microsoft/azure/cosmos/cassandra/CosmosDBLoadBalancingPolicy.java

```diff
@@ -0,0 +1,359 @@
+import com.google.common.collect.AbstractIterator;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.*;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+/**
+ * CosmosLoadBalancingPolicy allows you to specify readDC and writeDC to route read and write requests to their
+ * corresponding datacenters.
+ * globalEndpoint can also be used for writeDC. This allows the client to gracefully failover
+ * when the default write region is changed. dnsExpirationInSeconds is essentially the max duration to figure out the
+ * failover scenario. By default, it is 60 seconds.
+ * If readDC is specified, we prioritize nodes in the readDC for read requests.
+ * If writeDC is specified, we prioritize nodes in the writeDC for write requests.
+ */
+public class CosmosDBLoadBalancingPolicy implements LoadBalancingPolicy {
+
+    public static CosmosDBLoadBalancingPolicy buildFrom(Builder builder) {
+        validate(builder);
+        return new CosmosDBLoadBalancingPolicy(builder.readDC, builder.writeDC, builder.globalEndpoint, builder.dnsExpirationInSeconds);
+    }
+
+    /**
+     * Initializes the list of hosts in read, write, local, and remote categories.
+     */
+    @Override
+    public void init(Cluster cluster, Collection<Host> hosts) {
+        CopyOnWriteArrayList<Host> readLocalDCAddresses = new CopyOnWriteArrayList<Host>();
+        CopyOnWriteArrayList<Host> writeLocalDCAddresses = new CopyOnWriteArrayList<Host>();
+        CopyOnWriteArrayList<Host> remoteDCAddresses = new CopyOnWriteArrayList<Host>();
+
+        List<InetAddress> dnsLookupAddresses = new ArrayList<InetAddress>();
+        if (!globalContactPoint.isEmpty()) {
+            dnsLookupAddresses = Arrays.asList(getLocalAddresses());
+        }
+
+        for (Host host : hosts) {
+            if (!this.readDC.isEmpty() && host.getDatacenter().equals(readDC)) {
+                readLocalDCAddresses.add(host);
+            }
+
+            if ((!this.writeDC.isEmpty() && host.getDatacenter().equals(writeDC))
+                    || dnsLookupAddresses.contains(host.getAddress())) {
+                writeLocalDCAddresses.add(host);
+            }
+            else {
+                remoteDCAddresses.add(host);
+            }
+        }
+
+        this.readLocalDCHosts = readLocalDCAddresses;
+        this.writeLocalDCHosts = writeLocalDCAddresses;
+        this.remoteDCHosts = remoteDCAddresses;
+        this.index.set(new Random().nextInt(Math.max(hosts.size(), 1)));
+    }
+
+    /**
+     * Return the HostDistance for the provided host.
+     *
+     * <p>This policy consider the nodes for the writeDC and the default write region at distance {@code LOCAL}.
+     *
+     * @param host the host of which to return the distance of.
+     * @return the HostDistance to {@code host}.
+     */
+    @Override
+    public HostDistance distance(Host host) {
+        if (!this.writeDC.isEmpty()) {
```

Comment:
Why do we only consider write here?

Reply:
because that's where the main requests go (except reads)